### PR TITLE
Implement jwt_token_creds interop test

### PR DESCRIPTION
**Pull request by ejona86**
Using a JWT is a bit more work than it should be, but improving that
will come later.

At present, this test fails, but it is believed to be due to the auth
library.

### CODE REVIEW

File: interop-testing/src/main/java/io/grpc/testing/integration/TestServiceClient.java [hunk 1]
```diff
@@ -230,6 +231,10 @@ private void runTest(String testCase) throws Exception {
       String jsonKey = Files.toString(new File(serviceAccountKeyFile), Charset.forName("UTF-8"));
       FileInputStream credentialsStream = new FileInputStream(new File(serviceAccountKeyFile));
       tester.serviceAccountCreds(jsonKey, credentialsStream, oauthScope);
+    } else if ("jwt_token_creds".equals(testCase)) {
+      String jsonKey = Files.toString(new File(serviceAccountKeyFile), Charset.forName("UTF-8"));
```
Comment (madongfly) [inline interop-testing/src/main/java/io/grpc/testing/integration/TestServiceClient.java]: Unused?

File: interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java [hunk 2]
```diff
@@ -817,6 +818,29 @@ public void computeEngineCreds(String serviceAccount, String oauthScope) throws
     assertEquals(goldenResponse, response);
   }
 
+  /** Test JWT-based auth. */
+  public void jwtTokenCreds(InputStream serviceAccountJson) throws Exception {
```
Comment (madongfly) [inline interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java]: According to the [Spec](https://github.com/grpc/grpc/blob/master/doc/interop-test-descriptions.md#jwt_token_creds):
```
received SimpleResponse.username is not empty and is in the json key file used by the auth library. The client can optionally check the username matches the email address in the key file.
```
You may want to pass in the json key and do the check.

Comment (ejona86) [inline interop-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java]: I'm doing the "The client can optionally check the username matches the email address in the key file" which is more strict than the first part.

File: interop-testing/src/main/java/io/grpc/testing/integration/TestServiceClient.java (hunk 1 above)
Comment (ejona86) [inline interop-testing/src/main/java/io/grpc/testing/integration/TestServiceClient.java]: Removed.

**Comment by madongfly** [comment]
@ejona86 LGTM

**Comment by ejona86** [comment]
Cherry-picked as c68f9ff

**Pull request closed without merging**